src/test/appFormValidators.test.js

```diff
@@ -402,31 +403,81 @@ describe("App Form Validators", function () {
       it("port index to be valid", function () {
         expect(this.v.healthChecksPortIndex({
           protocol: "TCP",
+          portType: HealthCheckPortTypes.PORT_INDEX,
           portIndex: 1
         }))
           .to.be.true;
         expect(this.v.healthChecksPortIndex({
           protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_INDEX,
           portIndex: 0
         }))
           .to.be.true;
         expect(this.v.healthChecksPortIndex({
           protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_INDEX,
           portIndex: "abc"
         }))
           .to.be.false;
         expect(this.v.healthChecksPortIndex({
           protocol: "TCP",
+          portType: HealthCheckPortTypes.PORT_INDEX,
           portIndex: 0.5
         }))
           .to.be.false;
         expect(this.v.healthChecksPortIndex({
           protocol: "COMMAND",
+          portType: HealthCheckPortTypes.PORT_INDEX,
           portIndex: "abc"
         }))
           .to.be.true;
       });
 
+      it("port number to be a valid port", function () {
+        let healthChecksPort = this.v.healthChecksPort;
+
+        expect(healthChecksPort({
+          protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_NUMBER,
+          port: "NaN 666"
+        })).to.be.false;
+        expect(healthChecksPort({
+          protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_NUMBER,
+          port: -5
+        })).to.be.false;
+        expect(healthChecksPort({
+          protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_NUMBER,
+          port: 0.1
+        })).to.be.false;
+        expect(healthChecksPort({
+          protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_NUMBER,
+          port: "0.0001"
+        })).to.be.false;
+        expect(healthChecksPort({
+          protocol: "HTTP",
+          portType: HealthCheckPortTypes.PORT_NUMBER,
+          port: "70000"
```

Comment:
Minor nitpick: to prove the point `65536` (ie max value + 1) would have made things clear without feeling arbitrary!